src/pudl_rmi/validate.py

```diff
@@ -0,0 +1,178 @@
+        .astype({"report_year": pd.Int64Dtype()})
+        .groupby(by=by, dropna=True)[data_cols]
+        .sum(min_count=1)
+    )
+    return summed_out
+
+
+def agg_test_data(
+    df1: pd.DataFrame, df2: pd.DataFrame, data_cols: List[str], by: List[str], **kwarg
+) -> pd.DataFrame:
+    """
+    Merge two grouped input tables to determine if summed data column are equal.
+
+    Args:
+        df1: One dataframe to sum and check consistency with ``df2``.
+        df2: Other dataframe to sum and check consistency against ``df1``.
+        data_cols: data columns to check. Columns must be in both ``df1`` and
+            ``df2``.
+    """
+    test = pd.merge(
+        group_sum_cols(df1.pipe(add_data_source), data_cols=data_cols, by=by),
+        group_sum_cols(df2.pipe(add_data_source), data_cols=data_cols, by=by),
+        right_index=True,
+        left_index=True,
+        suffixes=("_1", "_2"),
+        how="outer",
+    ).astype("float64")
+    for data_col in data_cols:
+        test.loc[:, f"{data_col}_isclose"] = np.isclose(
+            test[f"{data_col}_1"],
+            test[f"{data_col}_2"],
+            equal_nan=True,
+            **kwarg,
+        )
+        test.loc[:, f"{data_col}_ratio"] = test[f"{data_col}_1"] / test[f"{data_col}_2"]
+        logger.info(
+            f"{data_col} close records: {len(test[test[f'{data_col}_isclose']])/len(test):.02%}"
+        )
+    return test.sort_index()
+
+
+def test_df_vs_net_plant_balance(
+    ferc_deprish_eia: pd.DataFrame,
+    pudl_engine: sa.engine.Engine,
+    data_cols: List = [
+        "plant_balance_w_common",
+        "book_reserve_w_common",
+        "unaccrued_balance_w_common",
+    ],
+    rtol: float = 5e-02,
+    atol: float = 5e-02,
+) -> pd.DataFrame:
+    """
+    Compare FERC-Deprish-EIA output to FERC1 Net Plant Balance totals.
+
+    The FERC1 Net Plant Balance table includes utility-FERC account level totals
+    of several key financial data points. This function compares the utility-FERC
+    account level ``data_cols`` from the Net Plant Balance table and the
+    FERC-Deprish-EIA outputs. To do this, we aggregate the more granular
+    FERC-Deprish-EIA output to the utility-FERC account level. Then we compare
+    the ``data_cols`` from each of  the two inputs using ``np.isclose``. We also
+    calculate a ratio of the FERC-Deprish-EIA column divided by the Net Plant
+    Balance column.
+
+    Question: Originally I passed in npb instead of generating it within this
+    function and just had a little wrapper function but that also felt bad. Any
+    Suggestions here?
+
+    Args:
+        ferc_deprish_eia: table of FERC-Deprish-EIA output.
+        pudl_engine: A connection engine for the PUDL DB.
+        data_cols: list of columns to compare.
+        rtol: The relative tolerance parameter from ``np.isclose``. Default is 5e-02.
+        atol: The absolute tolerance parameter from ``np.isclose``. Default is 5e-02.
```

Comment:
The default values should show up automatically in many / most contexts since they're specified above.

Reply:
grand!